src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    /// <summary>
+    /// A metadata class describing a required tag helper attribute.
+    /// </summary>
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        /// <summary>
+        /// The HTML attribute name.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The comparison method to use for <see cref="Name"/> when determining if an HTML attribute name matches.
+        /// </summary>
+        public TagHelperRequiredAttributeNameComparison NameComparison { get; set; }
+
+        /// <summary>
+        /// The HTML attribute value.
+        /// </summary>
+        public string Value { get; set; }
+
+        /// <summary>
+        /// The comparison method to use for <see cref="Value"/> when determining if an HTML attribute value matches.
+        /// </summary>
+        public TagHelperRequiredAttributeValueComparison ValueComparison { get; set; }
+
+        /// <summary>
+        /// Determines if the current <see cref="TagHelperRequiredAttributeDescriptor"/> matches the given
+        /// <paramref name="attributeName"/> and <paramref name="attributeValue"/>.
+        /// </summary>
+        /// <param name="attributeName">An HTML attribute name.</param>
+        /// <param name="attributeValue">An HTML attribute value.</param>
+        /// <returns><c>true</c> if the current <see cref="TagHelperRequiredAttributeDescriptor"/> matches 
+        /// <paramref name="attributeName"/> and <paramref name="attributeValue"/>; <c>false</c> otherwise.</returns>
+        public bool IsMatch(string attributeName, string attributeValue)
+        {
+            var nameMatches = false;
+            if (NameComparison == TagHelperRequiredAttributeNameComparison.FullMatch)
+            {
+                nameMatches = string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase);
+            }
+            else if (NameComparison == TagHelperRequiredAttributeNameComparison.PrefixMatch)
+            {
+                nameMatches = attributeName.Length != Name.Length &&
```

Comment:
Add a comment as I suggested earlier. Want future readers to know exact `!=` match is intentional.